draft-ietf-lwig-minimal-esp.xml

```diff
@@ -264,30 +264,33 @@ If application packets vary between 1 to 30 bytes, the application could always
-In addition, the Next Header may also carry an indication on how to process the packet <xref target="I-D.nikander-esp-beet-mode"/>.  </t>
+<section anchor="sec-nh" title="Next Header (8 bit) and Dummy Packets">
 
-<t> The ability to generate and receive dummy packets is required by <xref target="RFC4303"/>. 
-For interoperability, a minimal ESP implementation must discard dummy packets without indicating an error.
-Note that such recommendation only applies for nodes receiving packets, and that nodes designed to only send data might not implement this capability.  </t>
-      
-<t> As the generation of dummy packets is subject to local management and based on a per-SA basis, a minimal ESP implementation may not generate such dummy packet.
-Specifically, in constrained environment sending dummy packets may have too much impact on the device lifetime, and should be avoided. 
-On the other hand, constrained nodes may be dedicated to specific applications, in which case, traffic pattern may expose the application or the type of node. 
-For these nodes, not sending dummy packet may have some privacy implication that needs to be measured. 
-However, for the same reasons exposed in <xref target="sec-padding"/> traffic shaping at the IPsec layer may also introduce some traffic pattern, and on constrained devices the application is probably the most appropriated layer to limit the risk of leaking information by traffic shaping.</t>
-      
-<t> In some cases, devices are dedicated to a single application or a single transport protocol, in which case, the Next Header has a fixed value.</t> 
+<t>ESP <xref target="RFC4303"/> defines the Next Header as a mandatory 8 bits field in the packet.
+The Next header, only visible after decryption, specifies the data contained in the payload.
+In addition, the Next Header may also carry an indication on how to process the packet <xref target="I-D.nikander-esp-beet-mode"/>.
+The Next Header can point to a dummy packet, i.e. packets with the Next Header value set to 59 meaning "no next header".
+The data following to "no next header" is dummy data - unstructured random bytes that appear like real encrypted data.
+</t>
+
+<t> The ability to generate and to receive and ignore dummy packets is required by <xref target="RFC4303"/>.
+An implementation can omit ever generating and sending dummy packets.
+For interoperability, a minimal ESP implementation must process and discard dummy packets without indicating an error.
+</t>
+
+<t>
+In constrained environments, sending dummy packets may have too much impact on the device lifetime, in which case dummy packets should not be generated and sent.
+Constrained devices running specific applications that would leak too much information by not generating and sending dummy packets could implement this functionality instead at the application layer.
```

Comment:
Again, the glue of "On the other hand" is not needed. I removed those occurrences for readability. 
The other change is you adding "may implement this functionality or " which comes down to earlier discussions on not clearly defining a minimal model if you keep tagging all things to not do with "or you could just do it anyway".

I'm not sure what the "note" sentence at the end really adds. Especially the claim that dummy packets may also reveal some pattern". Can you explain how you think dummy packets could reveal something ?

Reply:
ok, that case makes sense. Why not include this to the description as an example. Maybe even add a line in the Security Considerations that Dummy packets work better on high speed links with lots of packets, and don't work very well on very low speed links that send packets of specific size on regular intervals, as it would require a right coupling between application data and IPsec stack code ?